WordPress/gutenberg
# Build: Update `npm run dev` watch task to a single call to watch function

## Description
This PR was an attempt to improve the performance of `npm run dev`. It doesn't seem to make much of a difference, but at least tidies up the code, and switches to using the node-watch API in a way that's closer to the author's intention (replacing the multiple calls to `watch` for each package with a single call that uses the `filter` property).

Also fixes a bug where the watcher wasn't deleting build files, the previous code was creating an incorrect path to the file to be deleted. It now works in this PR.

## How has this been tested?
1. Run `npm run dev`
2. Change some source code - the code should rebuild
3. Change some non-source code (e.g. test files) - the code should not rebuild
4. Delete a file - the deletion should be logged in the console and the corresponding build file removed.
